src/utils/buffer-helper.ts

```diff
@@ -146,4 +146,28 @@ export class BufferHelper {
     }
     return { len: bufferLen, start: bufferStart, end: bufferEnd, nextStart: bufferStartNext };
   }
+
+  // Trying to find buffered position within `maxFragLookUpTolerance`. Returns same position if
+  // current time is already inside of buffered range or there is no buffered fragment
+  // within `maxFragLookUpTolerance`.
+  static getBufferedPosWithinTolerance(media, pos, maxFragLookUpTolerance) {
+    const currentTime = media ? media.currentTime : undefined;
+
+    // No need to seek if current playback position is already inside buffered range
+    if (BufferHelper.isBuffered(media, currentTime)) {
+      return pos;
+    }
+
+    const buffered = media.buffered;
```

Comment:
maybe a style preference or for readability .. but wouldn't it be more performant not to create these variables and just use `media.buffered.length`  & `media.buffered.start(i)` directly? Just a suggestion, thanks